pkg/service/test/output.go

```diff
@@ -200,14 +200,64 @@ func diffIndex(s1, s2 string) (int, bool) {
  * minus or add symbol followed by the respective line.
  */
 func calculateJSONDiffs(json1 []byte, json2 []byte) (string, error) {
+
+	// Check if inputs are JSON arrays
+	if json1[0] == '[' && json1[len(json1)-1] == ']' && json2[0] == '[' && json2[len(json2)-1] == ']' {
+		json1 = []byte("{\"array\":" + string(json1) + "}")
+		json2 = []byte("{\"array\":" + string(json2) + "}")
+	}
+
+	var jsonObject1 map[string]interface{}
+	_ = json.Unmarshal(json1, &jsonObject1)
+
+	var jsonObject2 map[string]interface{}
+	_ = json.Unmarshal(json2, &jsonObject2)
+	returnedString := ""
+	// Check if jsonObject1 is an array
+	if array1, ok := jsonObject1["array"].([]interface{}); ok {
+		if array2, ok := jsonObject2["array"].([]interface{}); ok {
+			// Iterate over items in json1
+			for i, item1 := range array1 {
+				// Convert item to JSON bytes
+				itemJSON1, err := json.Marshal(item1)
+				if err != nil {
+					return "", err
+				}
+				item2 := array2[i]
+				itemJSON2, err := json.Marshal(item2)
+				if err != nil {
+					return "", err
+				}
+				// Perform comparison for each item
+				diffString, err := performComparison(itemJSON1, itemJSON2)
+				if err != nil {
+					return "", err
+				}
+				returnedString += diffString
+			}
+		}
```

Comment:
we internally used something similar thus iteration two arrays twice doesn't seem to be a good idea.

There exists alternatives which can ensure we don't have to compare manually, the issue I created was to look for such alternatives

Reply:
Ok I will search for it.

Reply:
@shivamsouravjha  We can use **go-test** package especially deep compare function 
https://pkg.go.dev/github.com/go-test/deep@v1.0.8
I made  a sample code to test it with two objects
`[
  {
    "firstName": "John",
    "lastName": "Doe",
    "address2": "14th Main Street, Royal Street",
    "city": "London, England",
    "telephone": "12345678",
    "id": 1,
    "pets": []
  },
  {
    "firstName": "Joe",
    "lastName": "Hopkins",
    "address": "24th Main Street, Something Street",
    "city": "Los Angeles, California",
    "telephone": "9090812030",
    "id": 2,
    "pets": []
  }
]
`
<hr>

`
[
  {
    "firstName2": "John",
    "lastName": "Doe",
    "address2": "14th Main Street, Royal Street",
    "city": "London, England",
    "telephone": "12345678",
    "id": 1,
    "pets": []
  },
  {
    "firstName": "Joe",
    "lastName": "Hopkins",
    "address": "24th Main Street, Something Street",
    "city": "Los Angeles, California",
    "telephone": "9090812030",
    "id": 2,
    "pets": []
  }
]
`
<hr>
I extracted the fields using regex and then append +, - before fields (it's a simulation about what will happen in keploy code).


`func TestDeepEqualJSONArrays() {`
`

	jsonArray1 := `[{"firstName":"John","lastName":"Doe","address2":"14th Main Street, Royal Street","city":"London, England","telephone":"12345678","id":1,"pets":[]},{"firstName":"Joe","lastName":"Hopkins","address":"24th Main Street, Something Street","city":"Los Angeles, California","telephone":"9090812030","id":2,"pets":[]}]`
	jsonArray2 := `[{"firstName2":"John","lastName":"Doe","address2":"14th Main Street, Royal Street","city":"London, England","telephone":"12345678","id":1,"pets":[]},{"firstName":"Joe","lastName":"Hopkins","address":"24th Main Street, Something Street","city":"Los Angeles, California","telephone":"9090812030","id":2,"pets":[]}]`

	var array1, array2 []map[string]interface{}

	// Unmarshal JSON arrays into slices of structs
	if err := json.Unmarshal([]byte(jsonArray1), &array1); err != nil {
		//print error
		fmt.Println("Error unmarshaling jsonArray1: %v", err)
	}
	if err := json.Unmarshal([]byte(jsonArray2), &array2); err != nil {
		fmt.Println("Error unmarshaling jsonArray1: %v", err)
	}


	// Compare arrays
	if diff := deep.Equal(array1, array2); diff != nil {
		re := regexp.MustCompile(`\[(\D\w*)\]:`)
		for _, d := range diff {

			if d[len(d)-1] == '>' {
				match := re.FindStringSubmatch(d)

				if len(match) > 1 {
					jsonArray1 = strings.Replace(jsonArray1, match[1], "+ "+match[1], 1)
				}
			} else {
				match := re.FindStringSubmatch(d)

				if len(match) > 1 {
					jsonArray2 = strings.Replace(jsonArray2, match[1], "- "+match[1], 1)
				}
			}
		}

		fmt.Println("Diff jsonArray1:", jsonArray1)
		fmt.Println("Diff jsonArray2:", jsonArray2)

	}
}
`

if it's good tell me to begin coding it in keploy